src/_pytest/stepwise.py

```diff
@@ -8,53 +9,70 @@
 from _pytest.main import Session
 from _pytest.reports import TestReport
 
+if TYPE_CHECKING:
+    from _pytest.cacheprovider import Cache
+
+STEPWISE_CACHE_DIR = "cache/stepwise"
+NO_PREVIOUSLY_FAILED_WONT_SKIP = "no previously failed tests, not skipping."
+PREVIOUSLY_FAILED_TEST_NOT_FOUND = "previously failed test not found, not skipping."
+SKIPPING_PASSED_ITEMS = "skipping {} already passed items."
+
 
 def pytest_addoption(parser: Parser) -> None:
     group = parser.getgroup("general")
     group.addoption(
         "--sw",
         "--stepwise",
         action="store_true",
+        default=False,
         dest="stepwise",
         help="exit on test failure and continue from last failing test next time",
     )
     group.addoption(
         "--stepwise-skip",
+        "--sw-skip",
         action="store_true",
+        default=False,
         dest="stepwise_skip",
         help="ignore the first failing test but stop on the next failing test",
     )
 
 
 @pytest.hookimpl
 def pytest_configure(config: Config) -> None:
-    config.pluginmanager.register(StepwisePlugin(config), "stepwiseplugin")
+    # We should always have a cache as cache provider plugin uses tryfirst=True
+    if config.option.stepwise:
+        config.pluginmanager.register(StepwisePlugin(config), "stepwiseplugin")
+
+
+def pytest_sessionfinish(session: Session) -> None:
+    config = session.config
+    assert config.cache is not None
+    if not config.option.stepwise:
+        # This hook exists so that --cache-show when empty will not output the empty [] for stepwise by default.
+        # Perhaps this is ok to set in the unconfigure hook?  But I added it here to avoid changing behaviour.
+        config.cache.set(STEPWISE_CACHE_DIR, [])
 
 
 class StepwisePlugin:
     def __init__(self, config: Config) -> None:
         self.config = config
-        self.active = config.getvalue("stepwise")
         self.session: Optional[Session] = None
         self.report_status = ""
-
-        if self.active:
-            assert config.cache is not None
-            self.lastfailed = config.cache.get("cache/stepwise", None)
-            self.skip = config.getvalue("stepwise_skip")
+        assert config.cache is not None
+        self.cache: Cache = config.cache
+        self.lastfailed: Optional[str] = self.cache.get(STEPWISE_CACHE_DIR, None)
+        self.skip: bool = config.option.stepwise_skip
```

Comment:
```suggestion
        self.skip: bool = config.getoption("stepwise_skip")
```

We never deprecated, but the "attr-access" to options through the `option` object is not recommended anymore, with `getoption` the recommended approach (doesn't really make a difference right now, but thought I would mention).